crates/milli/src/search/facet/filter.rs

```diff
@@ -230,6 +230,15 @@ impl<'a> Filter<'a> {
     pub fn evaluate(&self, rtxn: &heed::RoTxn<'_>, index: &Index) -> Result<RoaringBitmap> {
         // to avoid doing this for each recursive call we're going to do it ONCE ahead of time
         let filterable_fields = index.filterable_fields(rtxn)?;
+        for fid in self.condition.fids(MAX_FILTER_DEPTH) {
```

Comment:
`fids()` is doing extra traversal of the `Filter` tree every search, so wasn't sure about this approach. Better might be to gather attributes to be checked during `Filter` construction (in [`Filter::from_str`](https://github.com/meilisearch/meilisearch/blob/b21d7aedf9096539041362d438e973a18170f3fc/crates/milli/src/search/facet/filter.rs#L210)), though it looked like it could involve a lot of refactoring